checkr/src/interpreter.rs

```diff
@@ -151,7 +151,13 @@ impl AExpr {
                 }
             }
             AExpr::Binary(l, op, r) => op.semantic(l.semantics(m)?, r.semantics(m)?)?,
-            AExpr::Minus(n) => -n.semantics(m)?,
+            AExpr::Minus(n) => {
+                let temp = n.semantics(m);
+                match temp {
+                    Ok(val) => return val.checked_neg().ok_or(InterpreterError::ArithmeticOverflow),
+                    Err(e) => return Err(e),
```

Comment:
Instead of doing the matching by hand, we could use [`?`-operator](https://doc.rust-lang.org/book/ch09-02-recoverable-errors-with-result.html#a-shortcut-for-propagating-errors-the--operator) directly.

I think this would also allow us to inline the `temp` variable, and skip defining it.

Does that make sense?

Reply:
This is perfect, thanks! I'll just let CI finish and then I merge!

Reply:
Sounds good, I can see that I can reduce verbosity further, so I might just add one more commit, that makes it one line instead of having the curly brackets. I hope this is ok 🙂

Reply:
Whuups! I totally missed this comment before I merged. Feel free to open another small one.

Also, could you run `cargo fmt` now that you are at it?

Reply:
Thanks 🙂 I will get to this straight away

Reply:
The refactor should now be ready see #33 , `cargo fmt` was called on the code as well, so hopefully it is formatting compliant 🙂